### PR TITLE
Adds notification drop down to Nav Bar

### LINKED ISSUE #10: Add notifications dropdown to the nav-bar (UI Only)
screen shot 2017-04-22 at 10 51 31 pm

We need a similar dropdown to display notifications.

**What is expected ?**
- Notifications dropdown as a separate [component](https://guides.emberjs.com/v2.12.0/getting-started/core-concepts/#toc_components) added to the nav-bar.
- UI Only for now. 

**References:**
- https://guides.emberjs.com/v2.12.0/components/defining-a-component/
- https://semantic-ui.com/collections/menu.html

### PR DESCRIPTION
**Pull request by CosmicCoder96**
#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

#### Short description of what this resolves:

Adds a notification component to navbar (scrollable, in both mobile and large screens).

image

image

Fixes #10

### CODE REVIEW

File: app/styles/partials/popups.scss [hunk 1]
```diff
@@ -0,0 +1,7 @@
+.ui.notification.popup .list {
```
Comment (CosmicCoder96) [inline app/styles/partials/popups.scss]: Makes the pop up scrollable.

File: app/styles/partials/icons.scss [hunk 2]
```diff
@@ -1,3 +1,7 @@
 .compact.icon {
   margin: 0 !important;
 }
+
+.notification .icon {
```
Comment (CosmicCoder96) [inline app/styles/partials/icons.scss]: Makes the floating label appear closer to the notification icon

File: app/styles/partials/labels.scss [hunk 3]
```diff
@@ -0,0 +1,6 @@
+.notification .icon .label {
```
Comment (CosmicCoder96) [inline app/styles/partials/labels.scss]: Makes the floating label appear closer to the notification icon

**Comment by CosmicCoder96** [comment]
@niranjan94 Please review.

File: app/templates/components/nav-bar.hbs [hunk 4]
```diff
@@ -11,6 +11,12 @@
       <div class="results"></div>
     </div>
     <div class="right menu">
+      <a class="notification item">
```
Comment (niranjan94) [inline app/templates/components/nav-bar.hbs]: The icon + dropdown should be inside a component.

File: app/templates/components/nav-bar.hbs [hunk 5]
```diff
@@ -11,6 +11,12 @@
...
+        <i class="mail outline icon">
+          <div class="floating ui teal label">20</div>
```
Comment (niranjan94) [inline app/templates/components/nav-bar.hbs]: The font of the number seems to be weird. Could you check ?

File: app/templates/components/notification-dropdown.hbs [hunk 6]
```diff
@@ -0,0 +1,59 @@
+    <div class="ui wide notification popup bottom lef transition ">
```
Comment (niranjan94) [inline app/templates/components/notification-dropdown.hbs]: What does `lef` class mean ?

File: app/templates/components/notification-dropdown.hbs [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  <div class="ui fluid link  divided list">
+    <div class="item">
+      <div class="right floated content">
+        <a>{{t 'Mark all as Read'}}</a>
+      </div>
+    </div>
+    <div class="item">
+      <div class="content">
+        <a class="header"><span>New Session Propoal</span></a>
```
Comment (niranjan94) [inline app/templates/components/notification-dropdown.hbs]: No individual anchor elements inside a notification. The whole notification should be clickable. (See facebook for example)

File: app/templates/components/notification-dropdown.hbs [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        <div class="description">{{t 'Event'}} : <a class="item"><b>Sample</b></a></div>
```
Comment (niranjan94) [inline app/templates/components/notification-dropdown.hbs]: The description will just be plain text. 

And also, don't use translation helpers for dummy text. Doesn't make any sense.

File: app/components/notification-dropdown.js [hunk 9]
```diff
@@ -0,0 +1,7 @@
+import Ember from 'ember';
```
Comment (niranjan94) [inline app/components/notification-dropdown.js]: screen shot 2017-05-19 at 11 23 42 pm

- Why is there so much space everywhere ? 
- Why different widths on desktop ? Keep the same width as on mobile. (See how facebook does it) 
- Mark all read button seems to be mis-aligned and it looks the same as rest of the text. 
- See current implementation at https://eventyay.com

Have added more comments inline. 

Overall the UI looks _very_ unrefined.

**Comment by CosmicCoder96** [comment]
@niranjan94 Made some changes, please review again.

**Comment by niranjan94** [comment]
@CosmicCoder96 why was this PR closed? Always update existing PRs. When you close and open a new PR we loose the continuity to all the conversation that happened on the previous PR. Ideally, one issue should have one PR.

**Comment by CosmicCoder96** [comment]
@niranjan94 it wasn't on purpose. I accidentally force pushed while the PR was temporarily closed. And thus wasn't able to reopen it. Apologies.

**Pull request closed without merging**